thunder/dynamo/compiler.py

```diff
@@ -65,9 +453,17 @@ def __call__(self, gm: torch.fx.GraphModule, sample_args: list[torch.SymInt, tor
         # force recompilation of the GraphModule before passing it to Thunder.
         gm.real_recompile()
 
-        # Here in the future we could add some logic to check if the GraphModule
-        # is executable by Thunder, but for now we simply compile it and return
-        jitted_gm = jit(gm, **self.thunder_options)
-        self.thunder_fns.append(jitted_gm)
-        self.thunder_to_gm[jitted_gm] = gm
-        return jitted_gm
+        # Check if the complete graph `gm` is supported by thunder
+        # If yes, pass the whole `gm` to `thunder.jit` and return the compiled function.
+        # if is_graph_supported_by_thunder(gm, sample_args):
+        #     jitted_gm = self.thunder_jit(gm)
+        #     self.thunder_fns.append(jitted_gm)
+        #     self.thunder_to_gm[jitted_gm] = gm
+        #     compiled_fn = CompiledFunction(gm, jitted_gm, CompilerType.THUNDER)
+        #     self.subgraph_infos.append(SubgraphInfo(gm, [compiled_fn], False))
+        #     return jitted_gm
+
+        # The whole graph may not be supported by `thunder`, so we split it in `thunder` supported sections
+        # and unsupported sections which are passed to `torch.compile(backend='inductor')`
+        split_module = self.splitter(gm, sample_args)
```

Comment:
What happens if all operations are supported? Is a submodule still created? Do we need an option of bypassing the splitter?

Reply:
Yes, we create one submodule. This is for consistency in the returned `SubgraphInfo` object.